src/Features/CSharp/Portable/CodeRefactorings/PullMemberUp/Dialog/CSharpInterfacePullerViaDialog.cs

```diff
@@ -0,0 +1,53 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Composition;
+using Microsoft.CodeAnalysis.CodeGeneration;
+using Microsoft.CodeAnalysis.CodeRefactorings.PullMemberUp.Dialog;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Editing;
+using Microsoft.CodeAnalysis.Host.Mef;
+
+namespace Microsoft.CodeAnalysis.CSharp.CodeRefactorings.PullMemberUp.Dialog
+{
+    [ExportLanguageService(typeof(AbstractInterfacePullerWithDialog), LanguageNames.CSharp), Shared]
+    internal class CSharpInterfacePullerViaDialog : AbstractInterfacePullerWithDialog
+    {
+        protected override void ChangeMemberToPublicAndNonStatic(DocumentEditor editor, ISymbol symbol, SyntaxNode memberSyntax, SyntaxNode containingTypeNode, ICodeGenerationService codeGenerationService)
+        {
+            var modifier = DeclarationModifiers.From(symbol).WithIsStatic(false);
+            if (symbol is IEventSymbol eventSymbol)
+            {
+                if (memberSyntax.Parent != null &&
+                    memberSyntax.Parent.Parent is BaseFieldDeclarationSyntax eventDeclaration)
```

Comment:
can you extract out all the event handling into its own helper.  it's pretty confusing to try to figure out this entire method as is.

Reply:
OK